rust/processor/src/processors/token_v2_processor.rs

```diff
@@ -482,6 +482,7 @@ async fn parse_v2_token(
 
             // Get burn events for token v2 by object
             let mut tokens_burned: TokenV2Burned = AHashSet::new();
+            let mut tokens_burned_map: TokenV2BurnedMap = AHashMap::new();
```

Comment:
why do we need both?

Reply:
the hashset includes both old and new burn events. we still need to know if token got burnt with the old event for lookup. the map is for only new burn events to get the previous owner.

Reply:
we need to add comments here, and it might be better to merge them. E.g. for v1 event we can just have the value be null. It's very confusing to have both.